pkg/controllers/user/alert/watcher/node.go

```diff
@@ -197,32 +197,34 @@ func (w *NodeWatcher) checkNodeCPUUsage(alert *v3.ClusterAlert, machine *v3.Node
 
 func (w *NodeWatcher) checkNodeReady(alert *v3.ClusterAlert, machine *v3.Node) {
 	alertID := alert.Namespace + "-" + alert.Name
-	for _, cond := range machine.Status.InternalNodeStatus.Conditions {
-		if cond.Type == corev1.NodeReady {
-			if cond.Status != corev1.ConditionTrue {
-				clusterDisplayName := w.clusterName
-				cluster, err := w.clusterLister.Get("", w.clusterName)
-				if err != nil {
-					logrus.Warnf("Failed to get cluster for %s: %v", w.clusterName, err)
-				} else {
-					clusterDisplayName = cluster.Spec.DisplayName
-				}
-
-				data := map[string]string{}
-				data["alert_type"] = "nodeHealthy"
-				data["alert_id"] = alertID
-				data["severity"] = alert.Spec.Severity
-				data["alert_name"] = alert.Spec.DisplayName
-				data["cluster_name"] = clusterDisplayName
-				data["node_name"] = nodeHelper.GetNodeName(machine)
-
-				if cond.Message != "" {
-					data["logs"] = cond.Message
-				}
-				if err := w.alertManager.SendAlert(data); err != nil {
-					logrus.Debugf("Failed to send alert: %v", err)
+	if machine != nil {
+		for _, cond := range machine.Status.InternalNodeStatus.Conditions {
+			if cond.Type == corev1.NodeReady {
+				if cond.Status != corev1.ConditionTrue {
+					clusterDisplayName := w.clusterName
+					cluster, err := w.clusterLister.Get("", w.clusterName)
+					if err != nil {
+						logrus.Warnf("Failed to get cluster for %s: %v", w.clusterName, err)
+					} else {
+						clusterDisplayName = cluster.Spec.DisplayName
+					}
+
+					data := map[string]string{}
+					data["alert_type"] = "nodeHealthy"
+					data["alert_id"] = alertID
+					data["severity"] = alert.Spec.Severity
+					data["alert_name"] = alert.Spec.DisplayName
+					data["cluster_name"] = clusterDisplayName
+					data["node_name"] = nodeHelper.GetNodeName(machine)
+
+					if cond.Message != "" {
+						data["logs"] = cond.Message
+					}
+					if err := w.alertManager.SendAlert(data); err != nil {
+						logrus.Debugf("Failed to send alert: %v", err)
```

Comment:
should we increase the level to warn? or we don't care about alert being failed to send

Reply:
Because it will be trigger each 30 second, so we don't want too many logs.